library/core/src/char/methods.rs

```diff
@@ -1510,7 +1510,8 @@ impl char {
     #[rustc_const_stable(feature = "const_ascii_ctype_on_intrinsics", since = "1.47.0")]
     #[inline]
     pub const fn is_ascii_hexdigit(&self) -> bool {
-        matches!(*self, '0'..='9' | 'A'..='F' | 'a'..='f')
+        // Bitwise or can avoid need for branches in compiled code.
```

Comment:
nit: this comment says "bitwise" but the next line uses *logical* or.  I don't know which is correct, but it seems like it could be improved to be more consistent.

Reply:
The comment refers to the the bitwise or in `*self as u32 | 0x20`, which lets you check for uppercase and lowercase ASCII at once and therefore results in faster code. I can clarify the comment.